### PR TITLE
Feature/reading pm configuration data values

### PR DESCRIPTION
**Pull request by sponglord**
## Summary
PaymentMethod objects from the `/paymentMethods` response can now contain a configuration object. This object contains merchant defined values pulled from the CA.
This means that the merchant doesn't need to individually configure components when initialising Checkout - so PMs like GooglePay, PayPal & Korean cards can be given all the information they need to run in PayByLInk.
If the merchant does configure a component when initialising Checkout then these values will overwrite those from the `/paymentMethods` response

## Tested scenarios
Tested GooglePay, PayPal & Korean cards to ensure that configuration values from the PM object from the `/paymentMethods` response are read.
Tested that these values are overwritten by any locally defined merchant configuration

**Relates to issue**:  FOC-29226, FOC-32104 & DEV-29467

### CODE REVIEW

File: src/components/PayPal/types.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ declare global {
  * The intent for the transaction. This determines whether the funds are captured immediately, or later.
  * @see {@link https://developer.paypal.com/docs/checkout/reference/customize-sdk/#intent}
  */
-type Intent = 'sale' | 'capture' | 'authorize' | 'order';
+type Intent = 'sale' | 'capture' | 'authorize' | 'order' | '';
```
Comment (pabloai) [inline src/components/PayPal/types.ts:15]: I'd undo this change in order to make the `Intent` type more precise.
I know we have `intent: ''` in a couple of places, but we could replace the empty string with `null`.

Comment (sponglord) [reply src/components/PayPal/types.ts]: done

File: src/components/GooglePay/types.ts [hunk 2]
```diff
@@ -24,6 +28,7 @@ export interface GooglePayPropsConfiguration {
 export interface GooglePayProps extends UIElementProps {
     environment?: google.payments.api.Environment | string;
     configuration?: GooglePayPropsConfiguration;
+    paymentMethods?: PaymentMethod[];
```
Comment (marcperez) [inline src/components/GooglePay/types.ts:31]: Is there any reason why we added `paymentMethods` into the `GooglePayProps` types?

File: config/jest.config.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ module.exports = {
         '\\.scss$': '<rootDir>/config/testMocks/styleMock.js',
         '~(.*)$': '<rootDir>/src$1'
     },
-    verbose: true,
+    verbose: false,
```
Comment (marcperez) [inline config/jest.config.js:14]: Is this intentional? What is the motivation?

File: src/components/PayPal/Paypal.tsx [hunk 4]
```diff
@@ -25,6 +25,20 @@ class PaypalElement extends UIElement<PayPalElementProps> {
...
+    // Required for transition period (until configuration object becomes the norm)
+    // - if merchant has defined properties directly in props, use these instead
+    protected formatProps(props) {
+        const { configuration, intent, merchantId } = props;
+        return {
+            ...props,
+            configuration: {
+                ...configuration,
+                ...(intent && { intent }),
+                ...(merchantId && { merchantId })
```
Comment (pabloai) [inline src/components/PayPal/Paypal.tsx:37]: Shall we do a similar thing for cards? Since `koreanAuthenticationRequired` can be both inside and outside the `configuration` object.

File: config/jest.config.js (hunk 3 above)
Comment (sponglord) [reply config/jest.config.js:14]: If you want verbose testing turn it on - otherwise stop cluttering up the terminal

File: src/components/PayPal/Paypal.tsx (hunk 4 above)
Comment (sponglord) [reply src/components/PayPal/Paypal.tsx:37]: I already do:
```
configuration: {
                ...props.configuration,
                ...(props.koreanAuthenticationRequired !== undefined && { koreanAuthenticationRequired: props.koreanAuthenticationRequired })
            }
```

File: src/components/GooglePay/types.ts (hunk 2 above)
Comment (sponglord) [reply src/components/GooglePay/types.ts:31]: This can go - they were propagated in an earlier commit that was since revised

**Review by pabloai — APPROVED** [review]
